src/tsqlScheduler/Public/Remove-DatabaseTask.ps1

```diff
@@ -14,7 +14,12 @@ Function Remove-DatabaseTask {
 
 
     $deleteQuery = "
+set xact_abort on
+
 delete from scheduler.Task where TaskUid = '$($Task.TaskUid)'
+exec scheduler.DeleteAgentJob @taskUid = '$($Task.TaskUid)'
```

Comment:
This takes a pretty single-replica view of the world.  What happens to the other replicas?

The way agent jobs should get deleted is by setting IsDeleted to 1, and then the upsert job running.

Hard-deleting should be a rare occurrence - what's the specific flow you're aiming to solve here, as it might be that there should be extra handling, and hard deleting a job that isn't already logically deleted should fail.

Reply:
'what's the specific flow you're aiming to solve here' when syncing from folder to database to make the taskuids sync up. It deletes the old task, inserts the new task. but the new task can't create the new job because the job name exists from the old job.

Reply:
Is that a one-time activity, or is that expected to be a common case (someone deletes a task and adds a new one with a different `TaskUid` but identical name?).

For the rare case a custom script could take of this (and also ensure all the replicas have the old job deleted).

It's still possible for a job to exist on the server with a name collision - but I think your other change helps there by materializing the error.